src/Components/Blueprints/BuildImagesButton.tsx

```diff
@@ -1,36 +1,138 @@
+    value: string | number | undefined
+  ) => {
+    setIsOpen(false);
+  };
+    const {
+      data: blueprintDetails,
+    } =  useGetBlueprintQuery(selectedBlueprintId ? { id: selectedBlueprintId } : skipToken)
+    const blueprintImageType = blueprintDetails?.image_requests.map(
+        (image) => image.image_type
+    );
+
+  const onSelect2 = (
+      _event: React.MouseEvent<Element, MouseEvent> | undefined,
+      itemId: number | string | undefined
+  ) => {
+    const item = itemId as number;
+    const imageType = blueprintImageType?.[item];
+    if (selectedItems.includes(item)) {
+      setSelectedItems(selectedItems.filter((id) => id !== item));
+
+      setSelectedTargets(
+          selectedTargets.filter((target) => target !== imageType)
+      );
+    } else {
+      setSelectedItems([...selectedItems, item]);
+
+      if (imageType) {
+        setSelectedTargets([...selectedTargets, imageType]);
+      }
+    }
   };
-  return (
-    <Button
-      ouiaId="build-images-button"
-      onClick={onBuildHandler}
-      isDisabled={!selectedBlueprintId}
-      isLoading={imageBuildLoading}
-      variant={variant}
-    >
-      {children ? children : 'Build images'}
-    </Button>
-  );
+
+
+    return (
+      <Dropdown
+        isOpen={isOpen}
+        onSelect={onSelect}
+        onOpenChange={(isOpen: boolean) => setIsOpen(isOpen)}
+        toggle={(toggleRef: React.Ref<MenuToggleElement>) => (
+          <MenuToggle
+            variant="primary"
+            ref={toggleRef}
+            isFullWidth
+            onClick={onToggleClick}
+            isExpanded={isOpen}
+                splitButtonOptions={{
+                  variant: 'action',
+                  items: [
+                    <MenuToggleAction
+                        onClick={onBuildHandler}
+                        id="wizard-edit-save-btn"
+                        isDisabled={!selectedBlueprintId}
+                    >
+                      <Flex display={{default: 'inlineFlex'}}>
+                        {imageBuildLoading && (
+                            <FlexItem>
+                              <Spinner
+                                  style={
+                                    {'--pf-v5-c-spinner--Color': '#fff'} as React.CSSProperties
+                                  }
+                                  isInline
+                                  size="md"
+                              />
+                            </FlexItem>
+                        )}
+                        <FlexItem>Build images</FlexItem>
+                      </Flex>
+                    </MenuToggleAction>
+                  ]
+                }}
+            >
+          </MenuToggle>
+        )}
+      >
+        <Menu onSelect={onSelect2} selected={selectedItems}>
+          <MenuContent>
+            <MenuList>
+              {blueprintImageType?.map((imageType, index) => (
+                <MenuItem
+                  key={index}
```

Comment:
could we use the `imageType` as key and itemId and have selectedItems contain the imageType directly, not an index? I'm just thinking out loud, but might be bit safer (the order of blueprintImageType can not be relied on 100% IMO :) )

Reply:
All for using imageType as key since the values are unique :+1: